Feature/config icon

kagankan opened the pull request:
## What is the purpose of this PR?

- [ ] Fix bug
- [ ] Fix typo
- [ ] Update specs
- [ ] Add new rule
- [ ] Add new parser
- [ ] Improve or refactor rules
- [ ] Add a new core feature
- [ ] Improve or refactor core features
- [ ] Update documents
- [x] Others

### Description

Continued from #583 .
Add ⚙️ icon to code examples of configuration file.

- Changed the placement of the icon to take into account when the `title` is set.
- The shape of the curve is based on that used for OGP image and hero image.
- I only applied it to code examples represents the structure of the configuration file. 
  - I didn't apply it to code examples represents a part of the configuration file (e.g., represents only under `rules`).

For example, I'll apply it to:

```json
{
  "rules": {
    "invalid-attr": { "options": {} }
  }
}
```

I won't apply it to:

```json
{
  "invalid-attr": { "options": {} }
}
```


#### Screenshots

Without title:
![image](https://user-images.githubusercontent.com/34136752/219936890-650a910a-c4a2-4d2d-9d28-97bdb65d981f.png)

With title:
![image](https://user-images.githubusercontent.com/34136752/219936884-fac5ad2a-af65-4e42-bd5c-9a0a97a2de8e.png)


## Checklist

Fill out the checks for the applicable purpose.

- [ ] Fix bug
  - [ ] Add the test that can check whether resolved the issue.
- [ ] Update specs
  - [ ] Element
    - [ ] Update **[global attributes](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/html-spec/src/spec-common.attributes.json)**
    - [ ] Update **[content models](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/html-spec/src/spec-common.contents.json)**
    - [ ] Add also ARIA that you explored **ARIA in HTML [Recommendation](https://www.w3.org/TR/html-aria/) and [Editor's draft](https://w3c.github.io/html-aria/)**
  - [ ] Attribute
    - [ ] Update **[global attributes](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/html-spec/src/spec-common.attributes.json)**
    - [ ] Update **[IDL attribute](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/parser-utils/src/idl-attributes.ts)**
- [ ] Add new rule
  - [ ] Add a test
  - [ ] Add a README document
    - [ ] Create an issue that requests translating the doc if you want.
  - [ ] Add to [index](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/rules/src/index.ts)
  - [ ] Add JSON schema to [schema.json](https://github.com/markuplint/markuplint/blob/main/packages/%40markuplint/rules/schema.json)
  - [ ] Add to [`@markuplint/config-presets`](https://github.com/markuplint/markuplint/tree/main/packages/%40markuplint/config-presets)
    - [ ] And add to [Rulesets of base presets](https://markuplint.dev/docs/guides/presets#rulesets-of-base-presets)
      - [ ] Add to a [doc file](https://github.com/markuplint/markuplint/blob/main/website/docs/guides/presets.md)
      - [ ] Add to a [doc file (ja)](https://github.com/markuplint/markuplint/blob/main/website/i18n/ja/docusaurus-plugin-content-docs/current/guides/presets.md), or create an issue that requests translating the doc if you want.
  - [ ] Add to the [Playground](https://playground.markuplint.dev/)
    - [ ] Add to a [config object](https://github.com/markuplint/markuplint/blob/main/playground/src/ml-playground-home.svelte)
- [ ] Add new parser
  - [ ] Add a test
    - [ ] Do a test with some parser.
  - [ ] Add to [initializer](https://github.com/markuplint/markuplint/blob/main/packages/markuplint/src/cli/init/index.ts)

YusukeHirao commented on website/docs/guides/applying-rules.md:
````diff
@@ -76,7 +76,7 @@ Naturally, you can apply the [**custom rule**](./custom-rule) created by you or
 It applies the rule when specifying its plugin name and rule name solidus separated.
 
-```json title="Configuration"
+```json class=config title="Configuration" class=config
````
@kagankan They are duplicated.

YusukeHirao commented:
@kagankan 

All these need code block titles. 
Because these icons are "non-text content", and need an alternative text or context for each of them.

kagankan commented:
@YusukeHirao 
Thanks for your comment!

I recognized it as a decoration, but as you say, it indicates that the code is the configuration file...!
Fixed as follows:

- Fixed duplicated class attribute.
- Added **pseudo element's content** text. (not title)
  - Using pseudo element, we can show the text with CSS only.
  - If you want to output in the DOM, you will need to edit a lot of Markdown or React components.
- Update visual styles.
  - More area of colored background is too flashy.

![image](https://user-images.githubusercontent.com/34136752/221394001-0ec3690f-4f25-4545-bb9f-4a40403dd592.png)

## Question

Do you think pseudo element's content is acceptable as alternative text?
While understanding the advantages of the DOM, such as its searchability, I implemented it in CSS for ease of implementation.

kagankan commented:
If it should be a DOM text, I'll create a custom remark plugin (like npm2yarn plugin) 👍

YusukeHirao commented:
@kagankan 

> Do you think pseudo element's content is acceptable as alternative text?
> While understanding the advantages of the DOM, such as its searchability, I implemented it in CSS for ease of implementation.

Using DOM Text is more accessible, that is for sure, but no problem with using the pseudo-element because it is computed as an accessible name. I think copyable and searchable are not crucial to that text.
